### ISSUE TITLE
fatal error: 'longintrepr.h' file not found

### ISSUE THREAD

**Issue by henrydatei** [fabanc/pyvoronoi#22]
When installing with pip I get the above error:
```
Collecting pyvoronoi
  Using cached pyvoronoi-1.0.7.tar.gz (147 kB)
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: pyvoronoi
  Building wheel for pyvoronoi (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      Distribution mode: Compiling Cython generated .cpp sources.
      running bdist_wheel
      running build
      running build_ext
      building 'pyvoronoi' extension
      creating build
      creating build/temp.macosx-12-x86_64-cpython-311
      creating build/temp.macosx-12-x86_64-cpython-311/pyvoronoi
      clang -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -Ipyvoronoi -I/usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c pyvoronoi/pyvoronoi.cpp -o build/temp.macosx-12-x86_64-cpython-311/pyvoronoi/pyvoronoi.o
      pyvoronoi/pyvoronoi.cpp:198:12: fatal error: 'longintrepr.h' file not found
        #include "longintrepr.h"
                 ^~~~~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyvoronoi
  Running setup.py clean for pyvoronoi
Failed to build pyvoronoi
Installing collected packages: pyvoronoi
  Running setup.py install for pyvoronoi ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for pyvoronoi did not run successfully.
  │ exit code: 1
  ╰─> [16 lines of output]
      Distribution mode: Compiling Cython generated .cpp sources.
      running install
      /usr/local/lib/python3.11/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_ext
      building 'pyvoronoi' extension
      creating build
      creating build/temp.macosx-12-x86_64-cpython-311
      creating build/temp.macosx-12-x86_64-cpython-311/pyvoronoi
      clang -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -Ipyvoronoi -I/usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c pyvoronoi/pyvoronoi.cpp -o build/temp.macosx-12-x86_64-cpython-311/pyvoronoi/pyvoronoi.o
      pyvoronoi/pyvoronoi.cpp:198:12: fatal error: 'longintrepr.h' file not found
        #include "longintrepr.h"
                 ^~~~~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> pyvoronoi

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```
This error seems to occur with Python 3.11 (some other packages had this problem too after Python 3.11 came out)

**Comment by ghzeni** [comment]
I managed to install without receiving error messages using the inscrutions on the README.md:

![image](https://user-images.githubusercontent.com/15947263/229984054-b312e040-ded4-4299-9a84-71e6a1349ea3.png)

just had to run `pip3 install cython` first

**Comment by pnuaxen** [comment]
@ghzeni Can you confirm that you installed with Python 3.11?

**Comment by pnuaxen** [comment]
OK, I was also able to install in a python 3.11 virtual env and run it. However, I get warnings about the .egg being deprecated, so it's not a good long term solution

**Comment by RomainLaporte** [comment]
Is there any plan to officially support Python 3.11 and Python 3.12 ?

**Comment by h0h0h0** [comment]
Bumping here (apologies). Looking forward for support in 3.11 :)

**Comment by fabanc** [comment]
@h0h0h0  No worries. I just took control of the project again. I will try to reproduce and investigate in the next few days.

**Comment by fabanc** [comment]
@h0h0h0 @henrydatei @pnuaxen @ghzeni @RomainLaporte  I got the time to get back to the code. I created a specific branch to support 3.11. The problem you see is because the location of the file `longintrepr.h` has been moved from `include` to `include\cpython` in Python's installation folder.

I fixed the problem on windows at a release 1.0.8.1. I need to add the folder `include\cpython` to the list of folders passed to Microsoft's C++ compiler.

I need to do something similar for MacOs and Ubuntu. I am going to try to spin up a VM this week and dive into that. Once I have that working on Windows / Linux / Mac, I will release a format version 1.0.9 and merge the code into the main branch. 

Then I will move on to support 3.12.

**Comment by fabanc** [comment]
@henrydatei  Release 1.0.8.4 can be installed on Python 3.10, 3.11, tested on Windows 10 and Ubuntu 22.04.3. Can you give it a go on Mac? That's the only piece I cannot test myself. Something tell me the issue will still be there.

If the issue is still there, please take the code from this git repository, using the branch `Python3-11`. You will need to pip install `Cython` and `wheel`. 

Once you've done that, in the file setup.py, after line 82, add a line `print(c)`. Let me know the name of the compiler and I will try to pass it extra parameters just like what I am doing for windows. The error you had is similar to the one on Windows.

**Comment by h0h0h0** [comment]
@fabanc Nice dude! I'll stand by for the ubuntu release. let me know how i can help?

**Comment by fabanc** [comment]
Issue solved with commit #25 e1ae424. I have created a task so that the pypi project gets updated when I publish on GitHub, so `pip install pyvoronoi` will get you the latest version.

I have tested on Ubuntu from 3.9 to 3.12 using Docker. I have had a few Windows image testing 3.9, 3.11, 3.12. The only thing I could not test on is Mac OS. 

@h0h0h0 , I think I granted your wish. Go ahead and have fun! Note that the URL of the repository has changed since @Voxel8 transferred the ownership of that repo to me. It is good to keep that in mind.

**Comment by fabanc** [comment]
Issue solved with commit #25 e1ae424. I have created a task so that the pypi project gets updated when I publish on GitHub, so `pip install pyvoronoi` will get you the latest version.

I have tested on Ubuntu from 3.9 to 3.12 using Docker. I have had a few Windows image testing 3.9, 3.11, 3.12. The only thing I could not test on is Mac OS. 

@h0h0h0 , I think I granted your wish. Go ahead and have fun! Note that the URL of the repository has changed since @Voxel8 transferred the ownership of that repo to me. It is good to keep that in mind.
